src/backend/commands/tablecmds.c

```diff
@@ -11607,28 +11602,32 @@ static void checkUniqueIndexCompatible(Relation rel, GpPolicy *pol)
 		/* If the index is not a unique key, skip the check */
 		if (indexStruct->indisunique)
 		{
-			for (i = 0; i < indexStruct->indnatts; i++)
+			compatible = true;
+			if (indexStruct->indnatts < pol->nattrs)
+				compatible = false;
+
+			for (i = 0; i < pol->nattrs; i++)
```

Comment:
do we still need to loop here when `compatible`=`false`?
```suggestion
			for (i = 0; compatible && i < pol->nattrs; i++)
```

Reply:
No we don't need to, but I already have `break` below.